src/ray/rpc/gcs_server/gcs_rpc_server.h

```diff
@@ -62,7 +62,75 @@ class JobInfoGrpcService : public GrpcService {
+
+  virtual void HandleGetActorInfo(const GetActorInfoRequest &request,
+                                  GetActorInfoReply *reply,
+                                  SendReplyCallback send_reply_callback) = 0;
+
+  virtual void HandleRegisterActorInfo(const RegisterActorInfoRequest &request,
+                                       RegisterActorInfoReply *reply,
+                                       SendReplyCallback send_reply_callback) = 0;
+
+  virtual void HandleUpdateActorInfo(const UpdateActorInfoRequest &request,
+                                     UpdateActorInfoReply *reply,
+                                     SendReplyCallback send_reply_callback) = 0;
+};
+
+/// The `GrpcService` for `ActorInfoGcsService`.
+class ActorInfoGrpcService : public GrpcService {
+ public:
+  /// Constructor.
+  ///
+  /// \param[in] handler The service handler that actually handle the requests.
+  explicit ActorInfoGrpcService(boost::asio::io_service &io_service,
+                                ActorInfoHandler &handler)
+      : GrpcService(io_service), service_handler_(handler){};
+
+ protected:
+  grpc::Service &GetGrpcService() override { return service_; }
+
+  void InitServerCallFactories(
+      const std::unique_ptr<grpc::ServerCompletionQueue> &cq,
+      std::vector<std::pair<std::unique_ptr<ServerCallFactory>, int>>
+          *server_call_factories_and_concurrencies) override {
+    std::unique_ptr<ServerCallFactory> get_actor_info_call_factory(
+        new ServerCallFactoryImpl<ActorInfoGcsService, ActorInfoHandler,
+                                  GetActorInfoRequest, GetActorInfoReply>(
+            service_, &ActorInfoGcsService::AsyncService::RequestGetActorInfo,
+            service_handler_, &ActorInfoHandler::HandleGetActorInfo, cq, main_service_));
+    server_call_factories_and_concurrencies->emplace_back(
+        std::move(get_actor_info_call_factory), 1);
+
+    std::unique_ptr<ServerCallFactory> register_actor_info_call_factory(
+        new ServerCallFactoryImpl<ActorInfoGcsService, ActorInfoHandler,
+                                  RegisterActorInfoRequest, RegisterActorInfoReply>(
+            service_, &ActorInfoGcsService::AsyncService::RequestRegisterActorInfo,
+            service_handler_, &ActorInfoHandler::HandleRegisterActorInfo, cq,
+            main_service_));
+    server_call_factories_and_concurrencies->emplace_back(
+        std::move(register_actor_info_call_factory), 1);
+
+    std::unique_ptr<ServerCallFactory> update_actor_info_call_factory(
+        new ServerCallFactoryImpl<ActorInfoGcsService, ActorInfoHandler,
+                                  UpdateActorInfoRequest, UpdateActorInfoReply>(
+            service_, &ActorInfoGcsService::AsyncService::RequestUpdateActorInfo,
+            service_handler_, &ActorInfoHandler::HandleUpdateActorInfo, cq,
+            main_service_));
+    server_call_factories_and_concurrencies->emplace_back(
+        std::move(update_actor_info_call_factory), 1);
```

Comment:
can you define a macro (similar to RAY_CORE_WORKER_RPC_HANDLER) to simplify the boilerplate code?

Reply:
fixed